### PR TITLE
Fix `stable_baselines3/common/distributions` type hint

### PR DESCRIPTION
**Pull request by qgallouedec**
## Motivation and Context

- [ ] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [ ] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have reformatted the code using `make format` (**required**)
- [ ] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [ ] I have ensured `make pytest` and `make type` both pass. (**required**)
- [ ] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by qgallouedec** [comment]
I need feedback on these first changes. They are breaking, but the typing would be simpler, and the syntax more natural if entropy was always a tensor, and an error was raised when it has no analytic form.
Otherwise, we have to consider this case every time.

**Comment by araffin** [comment]
Look fine for me for now, I think the only place where it might be used is when using gSDE with PPO (so squash Gaussian with PPO/A2C).

File: stable_baselines3/common/policies.py [hunk 1]
```diff
@@ -628,7 +628,10 @@ def evaluate_actions(self, obs: th.Tensor, actions: th.Tensor) -> Tuple[th.Tenso
         distribution = self._get_action_dist_from_latent(latent_pi)
         log_prob = distribution.log_prob(actions)
         values = self.value_net(latent_vf)
-        entropy = distribution.entropy()
+        try:
+            entropy = distribution.entropy()
+        except NotImplementedError:
+            entropy = None
```
Comment (araffin) [inline stable_baselines3/common/policies.py:634]: we could directly approximate the entropy here, no?

Comment (qgallouedec) [reply stable_baselines3/common/policies.py:634]: I could be great, indeed.

File: stable_baselines3/common/policies.py [hunk 2]
```diff
@@ -631,7 +631,8 @@ def evaluate_actions(self, obs: th.Tensor, actions: th.Tensor) -> Tuple[th.Tenso
         try:
             entropy = distribution.entropy()
         except NotImplementedError:
-            entropy = None
+            # When no analytical form, entropy needs to be estimated using -log_prob.mean()
+            entropy = -log_prob.mean(dim=1)
```
Comment (araffin) [inline stable_baselines3/common/policies.py:635]: need to double check if it's mean or sum here

File: stable_baselines3/common/distributions.py [hunk 3]
```diff
@@ -567,11 +566,9 @@ def log_prob(self, actions: th.Tensor) -> th.Tensor:
             log_prob -= th.sum(self.bijector.log_prob_correction(gaussian_actions), dim=1)
         return log_prob
 
-    def entropy(self) -> Optional[th.Tensor]:
+    def entropy(self) -> th.Tensor:
         if self.bijector is not None:
-            # No analytical form,
-            # entropy needs to be estimated using -log_prob.mean()
-            return None
+            raise NotImplementedError("No analytical form, entropy needs to be estimated using -log_prob.mean()")
```
Comment (araffin) [inline stable_baselines3/common/distributions.py:571]: Note: do not forget to mentioned that as breaking change in the changelog

**Comment by araffin** [comment]
Closing in favor of https://github.com/DLR-RM/stable-baselines3/pull/1733

**Pull request closed without merging**